### DISCUSSION TITLE
role of "out" parameter in torch.matmul()

### DISCUSSION THREAD

**Discussion by sachinlodhi** [mrdbourke/pytorch-deep-learning#1114] [Q&A]
I was reading tensor [tutorial:](https://pytorch.org/tutorials/beginner/basics/tensorqs_tutorial.html#:~:text=Arithmetic%20operations,%3Dz3))
and found this optional parameter **out** in the torch.matmul(). Can please anyone tell what is the exact use of this parameter?
![image](https://github.com/user-attachments/assets/d544b03d-e153-42eb-8f67-afe2e4444c5e)

**Comment by Diwura** [comment]
hi @sachinlodhi  the "out" parameter specifies the tensor you want to store the result of the matrix multiplication. In the screenshot you posted, it would be save the matmul between tensor and tensor.T in y3. Instead of creating a new variable to hold the value of the matmul, you are doing it inline. I hope this helps.
